persistence/artist_repository.go

```diff
@@ -388,21 +407,16 @@ func (r *artistRepository) RefreshStats(allArtists bool) (int64, error) {
 		inClause := strings.Join(placeholders, ",")
 
 		// Replace the placeholder markers with actual SQL placeholders
-		batchSQL := strings.Replace(batchUpdateStatsSQL, "ROLE_IDS_PLACEHOLDER", inClause, 1)
-		batchSQL = strings.Replace(batchSQL, "TOTAL_IDS_PLACEHOLDER", inClause, 1)
-		batchSQL = strings.Replace(batchSQL, "UPDATE_IDS_PLACEHOLDER", inClause, 1)
-
-		// Create a single parameter array with all IDs (repeated 3 times for each IN clause)
-		// We need to repeat each ID 3 times (once for each IN clause)
-		var args []interface{}
-		for _, id := range artistIDBatch {
-			args = append(args, id) // For ROLE_IDS_PLACEHOLDER
-		}
-		for _, id := range artistIDBatch {
-			args = append(args, id) // For TOTAL_IDS_PLACEHOLDER
-		}
-		for _, id := range artistIDBatch {
-			args = append(args, id) // For UPDATE_IDS_PLACEHOLDER
+		batchSQL := strings.Replace(batchUpdateStatsSQL, "ROLE_IDS_PLACEHOLDER", inClause, 4)
+
+		// Create a single parameter array with all IDs (repeated 4 times for each IN clause)
+		// We need to repeat each ID 4 times (once for each IN clause)
+		args := make([]any, 4*len(artistIDBatch))
+		for idx, id := range artistIDBatch {
```

Comment:
IMHO, the old duplicated loop was more readable.

Reply:
(updated wrong thing). By old code, you mean 4 independent loops? A for loop 4 times?

Reply:
The way it was before, with 3 (now 4) independent loops

Reply:
Actually nvm.